esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,270 @@
+"""Part of the ESMValTool Aerosol diagnostics.
+
+This module contains utility functions commonly used by aerosol
+assessment routines.
+"""
+
+import os
+
+import iris
+import numpy as np
+
+
+class AeroAnsError(Exception):
+    pass
+
+
+# Global variables
+
+# STASH - name mapping file
+
+PATH_AERO = os.path.abspath(os.path.dirname(__file__))
+seasons = ['djf', 'mam', 'jja', 'son']
+nseas = len(seasons)
+mm_dust = 0.100  # Mol mass dust(kg mol-1) - see UKCA_MODE_SETUP: 'mm' arrays
+earth_g = 9.80665  # m/s^2
+
+
+def string_stc(stash):
+    """Convert stashcode to mXXsYYiZZZ format.
+
+    Parameters
+    ----------
+    stash : str
+        UM stashcode with format YYZZZ.
+
+    Returns
+    -------
+    : str
+        UM stashcode with format mXXsYYiZZZ.
+    """
+
+    stash = int(stash)
+
+    if stash < 1 or stash > 99999:
+        raise AeroAnsError('STR_STC: Invalid stashcode ' + str(stash))
+    return 'm01s{0:02d}i{1:03d}'.format(int(stash / 1000), int(stash % 1000))
+
+
+def add_bounds(cube):
+    """Add bounds to a cubes latitude and longitude coordinates.
+
+    Parameters
+    ----------
+    cube : Iris cube
+        Iris cube with latitude and longitude coordinates.
+
+    Returns
+    -------
+    cube : cube
+        Iris cube with bounds added to the latitude and longitude coordinates.
+    """
+
+    if not cube.coord('latitude').has_bounds():
+        cube.coord('latitude').guess_bounds()
+    if not cube.coord('longitude').has_bounds():
+        cube.coord('longitude').guess_bounds()
+
+    return cube
+
+
+def get_wavel_index(wavel):
+    """Return the pseudo-level index for given wavelength in AOD diags.
+
+    Parameters
+    ----------
+    wavel : str
+        A given wavelength.
+
+    Returns
+    -------
+    pseudo-level index : str
+        The pseudo-level index corresponding to the given wavelength.
+    """
+
+    # Wavelengths (nm)
+    cwavel = ['380', '440', '500', '675', '870', '1020']
+
+    try:
+        return cwavel.index(str(wavel)) + 1  # for 1: indexing in files
+
+    except AeroAnsError:
+        print('Get_Wavelen_Index: requested Wavelength not found' + str(wavel),
+              ' not in ', cwavel)
+
+
+def extract_pt(icube, pt_lat, pt_lon, **keys):
+    """Extracts given location(s) (3-D) from a cube.
+
+    Method
+    ------
+    Uses Iris module Analysis.Interpolate to extract values
+    first based on horizontal coordinates and later based on
+    height, if specified.
+
+    If height ('altitude') is requested, checks if cube heights
+    include orography, i.e. HybridHeights have been derived.
+
+    Parameters
+    ----------
+    icube : Iris cube
+    pt_lat, pt_lon : latitude and longitude coordinates of desired points
+    keys: (optional)
```

Comment:
I'd use `kwargs` here, and please specify the type of each kwarg/key; also, for better readability, you can always put those optional key word arguments inside the functional definition and initialize them with nulls eg `def extract_pt(icube, pt_lat, pt_lon, height=None, level=None, nearest=False):`

Reply:
I tried to initialize with nulls, but could not get the syntax right, so have only fixed this as per the above suggestion.

Reply:
nay bother, it's not a bug - it just depends on your preferred style :+1: